pkg/daemons/agent/agent.go

```diff
@@ -199,16 +200,33 @@ func CheckCgroups() (kubeletRoot, runtimeRoot string, hasCFS, hasPIDs bool) {
 	// For Unified (v2) cgroups we can directly check to see what controllers are mounted
 	// under the unified hierarchy.
 	if cgroupsModeV2 {
-		m, err := cgroupsv2.LoadManager("/sys/fs/cgroup", "/")
+
+		cgroupRoot, err := cgroupsv2.LoadManager("/sys/fs/cgroup", "/")
+		if err != nil {
+			logrus.Errorf("Failed to load root cgroup: %+v", err)
+			return "", "", false, false
+		}
+
+		cgroupRootProcs, err := cgroupRoot.Procs(false)
 		if err != nil {
 			return "", "", false, false
 		}
-		controllers, err := m.Controllers()
+
+		cgroupRootControllers, err := cgroupRoot.Controllers()
 		if err != nil {
 			return "", "", false, false
 		}
+
+		// evacuate from / to /init
+		if len(cgroupRootProcs) > 0 {
+			if err := cgrouputil.EvacuateCgroup2("init"); err != nil {
```

Comment:
This should be probably called outside of CheckCgroups(). Or rename CheckCgroups(), as this is now more than “checking”

Reply:
:heavy_check_mark: